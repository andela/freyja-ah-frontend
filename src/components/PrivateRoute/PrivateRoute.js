import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

export class PrivateRoute extends React.Component {
  constructor(props) {
    super(props);
    this.state = { };
  }

  render() {
    const { component: Component, isAuthenticated, ...rest } = this.props;
    return (
      <Route
        {...rest}
        render={props => (isAuthenticated ? <Component {...props} />
          : <Redirect to={{ pathname: '/login', state: { referrer: props.location } }} />)}
      />
    );
  }
}


PrivateRoute.propTypes = {
  component: PropTypes.elementType,
  isAuthenticated: PropTypes.bool,
};
const mapStateToProps = state => ({
  isAuthenticated: state.auth.isAuthenticated,
});

export default connect(mapStateToProps)(PrivateRoute);
