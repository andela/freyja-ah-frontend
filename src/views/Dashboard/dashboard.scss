.d-container {
  width: 100%;
}

.m-main{
  margin-top:20px;
}

div .d-row {
    margin-left: 2%;
}
div.s-row {
  margin-right: 0;
}
.d-aside{
  padding-top: 2rem;
}
.d-row .row {
  padding-left: 4rem;
}

 .d-sidebar aside {
  height: 100%;
  padding: 10px;
}

.d-links li.module-active {
  background: $sidebar-links-color;
  &:hover {
    border: 0;
  }
}

.d-links li.module-active a {
  color: $white;
}
.d-links li a {
  text-decoration: none;
  color: $sidebar-links-color;
}

.footer-div.d-footer {
  position: unset;
}
ul.d-links {
  list-style: none;
  margin: 0;
  padding: 0;
}

.d-links li, span.d-logo {
  cursor: pointer;
  margin: 4rem 1rem;
  text-align: center;
  padding: 5px;
}

.d-links li:hover {
  border: 2px solid $sidebar-links-color;
  border-radius: $s-radius;
}

a.m-card-link {
  color: $black;
}
