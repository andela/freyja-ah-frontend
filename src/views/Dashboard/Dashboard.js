/* eslint-disable react/prefer-stateless-function */
import React, { Component, Fragment } from 'react';
import { Row, Col } from 'reactstrap';
import { bindActionCreators } from 'redux';
import { withRouter, Link } from 'react-router-dom';
import { func, arrayOf, object } from 'prop-types';
import { connect } from 'react-redux';
import NavBar from '../../components/Header/Header';
import Footer from '../../components/Footer/Footer';
import Modules from '../../components/Modules/Modules';
import Sidebar from '../../components/Sidebar/Sidebar';
import { getModules } from '../../store/actions/moduleActions';
import './dashboard.scss';
import sideLogo from '../../assets/images/logo.png';

/**
 * @class SignUp
 * @description SignUp component
 * @param {object} event - Synthetic event object
 */
export class Dashboard extends Component {
  componentDidMount() {
    const { getCourseModules } = this.props;
    getCourseModules();
  }

  render() {
    const { allModules } = this.props;
    return (
      <Fragment>
        <main className="d-container">
          <NavBar />
          <div className="d-row">
            <Row className="s-row">
              <Col sm="2" className="d-sidebar">
                <Sidebar>
                  <div className="d-aside">
                    <div>
                      <span className="d-logo">
                        <img src={sideLogo} alt="logo" />
                        Community
                      </span>
                    </div>
                    <ul className="d-links">
                      <li className="module-active"><Link to="/dashboard">Modules</Link></li>
                      <li><Link to="/profile">Profile</Link></li>
                      <li><Link to="/payment">Payment</Link></li>
                      <li><Link to="/aboutus">Help</Link></li>
                    </ul>
                  </div>
                </Sidebar>
              </Col>
              <Col sm="10" className="m-main">
                <Modules modules={allModules} />
              </Col>
            </Row>
          </div>
          <div className="footer-div d-footer">
            <Footer />
          </div>
        </main>

      </Fragment>
    );
  }
}

/**
 * @method mapDispatchToProps
 * @description maps redux actions to props
 * @param {callback} dispatch destructured reducer state object
 * @returns {object} state
 */
export const mapDispatchToProps = dispatch => bindActionCreators(
  {
    getCourseModules: getModules,
  },
  dispatch,
);

/**
 * @method mapStateToProps
 * @description maps reducer states to props
 * @param {object} * destructured reducer state object
 * @returns {object} state
 */
export const mapStateToProps = ({ modules }) => {
  const {
    allModules,
    errors,
  } = modules;
  return {
    allModules,
    errors,
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(withRouter(Dashboard));

Dashboard.propTypes = {
  getCourseModules: func.isRequired,
  allModules: arrayOf(object),
};
