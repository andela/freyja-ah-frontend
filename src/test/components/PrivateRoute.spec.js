import React from 'react';
import { shallow } from 'enzyme';
import { PrivateRoute } from '../../components/PrivateRoute/PrivateRoute';
import { Dashboard } from '../../views/Dashboard/Dashboard';

const props = {
  component: Dashboard,
  isAuthenticated: true,
};
describe('component: PrivateRoute', () => {
  it('should render correctly', (done) => {
    const wrapper = shallow(<PrivateRoute />);
    expect(wrapper).toMatchSnapshot();
    done();
  });
  it('should render when user is Authenticated', (done) => {
    const wrapper = shallow(<PrivateRoute {...props} />);
    expect(wrapper).toMatchSnapshot();
    done();
  });
});
