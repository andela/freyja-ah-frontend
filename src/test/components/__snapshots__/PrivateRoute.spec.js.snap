// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: PrivateRoute should render correctly 1`] = `ShallowWrapper {}`;

exports[`component: PrivateRoute should render when user is Authenticated 1`] = `ShallowWrapper {}`;
